Dockerfile

```diff
@@ -1,12 +1,59 @@
+##
+# Build Stage
+#
+FROM maven:3-jdk-11 as build
+
+RUN apt-get -y update
+RUN apt-get -y install git apache2-utils
+
+##
+# Copy Source Code
+#
+ADD .git /build/.git
+ADD .gitmodules /build/.gitmodules
+ADD ./pom.xml /build/pom.xml
+ADD ./tools /build/tools
+ADD ./src /build/src
+ADD ./install /build/install
+ADD ./dev /build/dev
+
+WORKDIR /build
+
+##
+# Checkout Dependencies
+#
+RUN git submodule update --init --recursive
+
+ENV DEFAULT_PW="adminADMIN!"
```

Comment:
@pmanko , i think one remaining minor issue ,
we would better pass this as ARG , so that it can easily be passed to the builder action.
ENVs are more suitable for  variables needed at runtime , rather than at Build time